### ISSUE TITLE
bionic-ppc64el autopkgtest boot failure

### ISSUE THREAD

**Issue by gcoretech** [systemd/systemd#17250]
In case nobody noticed yet, bionic-ppc64el autopkgtest seems to fail because the virtual machine does not start.

See http://autopkgtest.ubuntu.com/running#pkg-systemd-upstream (search for ALERT)
```
QEMU Starting
Build Date = Dec 13 2017 13:46:58
FW Version = buildd@ release 20170724
Press "s" to enter Open Firmware.
...
Booting from memory...
HV-RTAS-UPDATE error: -4 
...
Linux ppc64le
#119-Ubuntu SMP Loading, please wait...
Starting version 246.1060.g5b36e8f2e4.0
Failed to write 'add' to '/sys/devices/vio/uevent': No such device
Begin: Loading essential drivers ... done.
Begin: Running /scripts/init-premount ... done.
Begin: Mounting root file system ... Begin: Running /scripts/local-top ... done.
Begin: Running /scripts/local-premount ... Scanning for Btrfs filesystems
done.
Begin: Waiting for root file system ... Begin: Running /scripts/local-block ... done.
done.
Gave up waiting for root file system device.  Common problems:
 - Boot args (cat /proc/cmdline)
   - Check rootdelay= (did the system wait long enough?)
 - Missing modules (cat /proc/modules; ls /dev)
ALERT!  /dev/sda1 does not exist.  Dropping to a shell!
```

**Comment by mrc0mmand** [comment]
This is out of my realm, unfortunately. For Ubuntu CI, you'll have to contact either @xnox or @ddstreet .

**Comment by ddstreet** [comment]
Ah, my guess would be this is a result of 88da55e28b467999da005591d3252a98f4436522 since on ppc64el the vio driver is buggy and returns ENODEV for the /sys/device/vio node:
https://github.com/systemd/systemd/issues/13652#issuecomment-535129791

I made a small effort to fix that in the kernel, but never followed up.  I can try again to fix the kernel but for the short term, maybe ENODEV should only log error but not completely exit out of the trigger loop? I don't think ENODEV would indicate use by an unprivileged user, would it?

@yuwata what do you think about allowing the loop to continue on ENODEV instead of returning error? I should note that I haven't actually verified this is what's causing the ubuntu-ci failure on ppc64el, it's just my first guess; I'll give it a test to try to verify it today.

**Comment by yuwata** [comment]
Ouch, fix is waiting in #17253.
